GameFighter/Battle.cs

```diff
@@ -19,13 +19,24 @@ public static bool Fight(Warrior warrior1, Warrior warrior2)
 
             while(warrior1.IsAlive)
             {
-                warrior2.GetAttack(warrior1);
+                int actualattack = warrior2.GetAttack(warrior1);
+
+                if (warrior1 is Vampire)
```

Comment:
This logic shouldn't be placed here. Fight class isn't responsible for handling different warrior types.
You need yo carefully use _as_ and _is_ statements. Read this article https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast
The general idea is that when you use these operators they perform casting. This operation requires some resources. In your case there are 2 casings: when you use _is_ operator and _as_. It's not efficient